app/renderer/components/menubar.js

```diff
@@ -102,193 +100,69 @@ class Menubar extends ImmutableComponent {
-
-          let openedSubmenuDetails = this.props.contextMenuDetail.get('openedSubmenuDetails')
-            ? this.props.contextMenuDetail.get('openedSubmenuDetails')
-            : new Immutable.List()
-
-          const rect = this.getContextMenuItemBounds()
-          const itemHeight = (rect.bottom - rect.top)
-
-          openedSubmenuDetails = openedSubmenuDetails.push(Immutable.fromJS({
-            y: (rect.top - itemHeight),
-            template: selectedMenuItem.get('submenu')
-          }))
-
-          windowActions.setContextMenuDetail(this.props.contextMenuDetail.set('openedSubmenuDetails', openedSubmenuDetails))
-          break
-        }
+        e.preventDefault()
+        e.stopPropagation()
 
         // Regular old menu item
         const nextIndex = selectedIndex === null
           ? 0
           : wrappingClamp(
-              selectedIndex[0] + (e.which === keyCodes.LEFT ? -1 : 1),
+              selectedIndex + (e.which === keyCodes.LEFT ? -1 : 1),
               0,
               this.props.template.size - 1)
 
-        // Context menu already being displayed; auto-open the next one
+        windowActions.setMenuBarSelectedIndex(nextIndex)
+
         if (this.props.contextMenuDetail) {
-          windowActions.setSubmenuSelectedIndex([nextIndex, 0])
-          showContextMenu(this.getMenubarItemBounds(nextIndex), this.getTemplate(nextIndex).toJS(), this.props.lastFocusedSelector)
+          windowActions.setContextMenuSelectedIndex([0])
+          showContextMenu(this.getMenubarItemBounds(nextIndex), template.get(nextIndex).get('submenu').toJS(), this.props.lastFocusedSelector)
         } else {
-          windowActions.setSubmenuSelectedIndex([nextIndex])
+          windowActions.setContextMenuSelectedIndex()
         }
         break
 
-      case keyCodes.UP:
       case keyCodes.DOWN:
-        e.preventDefault()
-        if (this.getTemplateItemsOnly()) {
-          if (!this.props.contextMenuDetail) {
-            // First time hitting up/down; popup the context menu
-            const newIndices = selectedIndex.slice()
-            newIndices.push(0)
-            windowActions.setSubmenuSelectedIndex(newIndices)
-            showContextMenu(this.getMenubarItemBounds(), this.getTemplate().toJS(), this.props.lastFocusedSelector)
-          } else {
-            // Context menu already visible; move selection up or down
-            const nextIndex = wrappingClamp(
-              this.currentIndex + (e.which === keyCodes.UP ? -1 : 1),
-              0,
-              this.maxIndex)
-
-            const newIndices = selectedIndex.slice()
-            if (this.hasMenuSelection) {
-              newIndices[selectedIndex.length - 1] = nextIndex
-            } else {
-              newIndices.push(0)
-            }
-            windowActions.setSubmenuSelectedIndex(newIndices)
-          }
+      case keyCodes.ENTER:
+        const contextMenuIndex = windowStore.getState().getIn(['ui', 'contextMenu', 'selectedIndex'])
```

Comment:
same as comment above; would be good to remove reference to windowStore and then bind it using a property in `main.js`